crates/pgt_workspace/src/workspace/server/statement_identifier.rs

```diff
@@ -35,91 +17,83 @@ impl From<usize> for RootId {
 /// ```
 ///
 /// For now, we only support SQL functions – no complex, nested statements.
-///
-/// An SQL function only ever has ONE child, that's why the inner `RootId` of a `Root`
-/// is the same as the one of its `Child`.
 pub enum StatementId {
-    Root(RootId),
-    // StatementId is the same as the root id since we can only have a single sql function body per Root
-    Child(RootId),
-}
-
-impl Default for StatementId {
-    fn default() -> Self {
-        StatementId::Root(RootId { inner: 0 })
-    }
+    Root {
+        content: Arc<str>,
+    },
+    Child {
+        content: Arc<str>,        // child's actual content
+        parent_content: Arc<str>, // parent's content for lookups
+    },
 }
 
 impl StatementId {
-    pub fn raw(&self) -> usize {
-        match self {
-            StatementId::Root(s) => s.inner,
-            StatementId::Child(s) => s.inner,
+    pub fn new(statement: &str) -> Self {
+        StatementId::Root {
+            content: statement.into(),
         }
     }
 
-    pub fn is_root(&self) -> bool {
-        matches!(self, StatementId::Root(_))
-    }
-
-    pub fn is_child(&self) -> bool {
-        matches!(self, StatementId::Child(_))
+    /// Creates a child statement ID with the given content and parent content.
+    pub fn new_child(child_content: &str, parent_content: &str) -> Self {
+        StatementId::Child {
+            content: child_content.into(),
+            parent_content: parent_content.into(),
+        }
```

Comment:
Is this used/needed? feels like we could create orphaned childs with this – not a problem right now, but maybe leads to confusion in the future?

Reply:
nope, thanks!